engine/class_modules/monk/sc_monk.cpp

```diff
@@ -2061,6 +2086,36 @@ struct blackout_kick_t : charred_passions_t<monk_melee_attack_t>
 // Rushing Jade Wind
 // ==========================================================================
 
+struct flight_of_the_red_crane_dmg_t : public monk_spell_t
+{
+  flight_of_the_red_crane_dmg_t( monk_t *p )
+    : monk_spell_t( p, "flight_of_the_red_crane_dmg",
+                    p->talent.conduit_of_the_celestials.flight_of_the_red_crane_dmg )
+  {
+    background = true;
+    aoe        = p->talent.conduit_of_the_celestials.flight_of_the_red_crane->effectN( 1 ).base_value();
+  }
+
+  void execute() override
+  {
+    monk_spell_t::execute();
+
+    p()->buff.flight_of_the_red_crane->trigger();
+  }
+};
+
+struct flight_of_the_red_crane_heal_t : public monk_heal_t
+{
+  flight_of_the_red_crane_heal_t( monk_t *p )
+    : monk_heal_t( p, "flight_of_the_red_crane_heal",
+                   p->talent.conduit_of_the_celestials.flight_of_the_red_crane_heal )
+  {
+    background = true;
+    aoe        = p->talent.conduit_of_the_celestials.flight_of_the_red_crane->effectN( 1 ).base_value();
+    target     = p;
+  }
+};
+
```

Comment:
I would template this much like the current implementation of Chi Wave/Chi Burst.

`flight_of_the_red_crane_heal_t` is likely missing an `execute()` override identical to the one found in `flight_of_the_red_crane_dmg_t`.

This style would apply to all CoC Celestial summons, I believe.